src/main.rs

```diff
@@ -27,14 +27,7 @@ fn main() -> Result<()> {
             nix::lock::nix_flake_lock_check(&cfg.sub_flake_url(&url))?;
         }
 
-        let outs = nix::devour_flake::devour_flake(args.verbose, nix_args)?;
-        if outs.len() == 0 {
-            eprintln!("Warn: devour-flake produced no outputs")
-        } else {
-            for out in outs {
-                println!("{}", out.0.bold());
-            }
-        }
```

Comment:
This code should not have been removed?

Reply:
`nixci` is expected to produce the outputs in STDOUT. That's why there is a `println!` here, and everywhere else we use `eprintln!`. 

This behaviour is relied by others, such as this Jenkins module: https://github.com/juspay/jenkins-nix-ci/blob/cd38c321b7d4b4e7621ec62bfdbf6a4c37028d30/nix/jenkins/features/nix/nixCI.groovy#L13

Reply:
This was inline into `print_devour_flake_output` (previously called `parse_devour_flake_output`). Printing lots from `nix build` can get noisy (I like `nix build -L` :wink: ) so this avoids having to allocate/buffer all the lines up front

Reply:
Oh n/m. I see `print_devour_flake_output` below.

Reply:
> Printing lots from `nix build` can get noisy (I like `nix build -L` 😉 ) so this avoids having to allocate/buffer all the lines up front

I don't understand. devour-flake outputs only the built outputs in the stdout, so it doesn't matter if using `-L`. And these outputs are read entirely into memory anyway (see `raw_output: Vec<u8>` argument in your `print_devour_flake_output`).

This function's (`devour_flake`) interface has also regressed since it no longer returns a vector of `DrvOut` which type is also missing now. I'd revert this change, and then your PR is good to merge as I've now tested it locally.

Reply:
Oh that's a good point, maybe I've mixed things up in my mind so maybe this optimization isn't actually necessary...

I had noticed that `nixci` was hanging silently at some point and assumed it was due to output buffering, but later I realized it was hanging due to me having Ctrl-C'd an instance earlier (seems like Ctrl-C doesn't kill the `nix build` under the hood)

Reply:
But yeah feel free to revert!

Reply:
This should be fixed!